lib/rbreadline.rb

```diff
@@ -4740,6 +4748,7 @@ def _rl_internal_char_cleanup()
   def readline_internal_charloop()
     lastc = -1
     eof_found = false
+    @is_pipe = (!@rl_outstream.winsize rescue true)
```

Comment:
Do you think it could work if we checked `!rl_outstream.tty?`, that way we don't introduce the `io/console` dependency...

Reply:
> Do you think it could work if we checked !@rl_outstream.tty?

I checked every reasonably obvious means of testing for it in core, and I'm sure I tried `tty`.  Nothing worked.

But, I should check in [`_rl_get_screen_size`](https://github.com/ConnorAtherton/rb-readline/blob/78f0f7a9ba3c4ab32d9c95e3897a5b1c8f82d152/lib/rbreadline.rb#L1859-L1867), as it might show the same info.

> we don't introduce the io/console dependency...

I'm not bothered about a std-lib/default gem dependency, given that `rb-readline` isn't turning away collaborators.  If this repo can make use of c code wrapped there, less support required here.